napari/_vispy/_tests/test_vispy_labels_layer.py

```diff
@@ -95,3 +96,26 @@ def test_labels_painting_with_mouse(
     assert not np.any(visual.node._data)
     mouse_press_callbacks(layer, event)
     assert np.any(visual.node._data)
+
+
+@skip_local_popups
+@skip_on_win_ci
+def test_labels_iso_gradient_modes(make_napari_viewer):
+    """Check that we can set `iso_gradient_mode` with `iso_categorical` rendering (test shader)."""
+    # NOTE: this test currently segfaults on Windows CI, but confirmed working locally
+    # because it's a segfault, we have to skip instead of xfail
+    viewer = make_napari_viewer(show=True)
+    labels = make_labels_layer('numpy', shape=(32, 32, 32))
+    labels[14:18, 14:18, 14:18] = 1
+    layer = viewer.add_labels(labels)
+    viewer.dims.ndisplay = 3
+    QCoreApplication.instance().processEvents()
+    assert layer.rendering == 'iso_categorical'
+
+    layer.iso_gradient_mode = IsoCategoricalGradientMode.SMOOTH
+    QCoreApplication.instance().processEvents()
+    assert layer.iso_gradient_mode == 'smooth'
+
+    layer.iso_gradient_mode = IsoCategoricalGradientMode.FAST
+    QCoreApplication.instance().processEvents()
+    assert layer.iso_gradient_mode == 'fast'
```

Comment:
I do not see that this test requires a whole viewer. May you change it to use `QtViewer` and `ViewerModel`? 

```python 
    viewer = ViewerModel()
    qt_viewer = QtViewer(viewer) 
    qtbot.add_widget(qt_viewer)
```

This should speed up tests and remove the fragility of our `make_napari_viewer` fixture teardown. 

It will be nice to check `qt_viewer.layer_to_visual[labels].iso_gradient_mode` value

Reply:
I am happy to change this, but can you explain a bit or point to documentation where I can learn which tests may require `Viewer` (`make_napari_viewer`) vs. `QtViwer` and `ViewerModel`? All other tests in this file for example use `make_napari_viewer`.

I also would not have thought to import from `napari._qt` in this file - that should be okay?

Reply:
`QtViewer` is importable from `napari.qt` 

We may use `ViewerModel` + `QtViewer` in any test that does not need access to `viewer.window`. 

There may be a need to show (`qt_viewer.show()`) widget to trigger rendering.

Reply:
That makes sense - thanks! It seems there may be a number of tests that unnecessarily use `make_napari_viewer` in that case.

Reply:
I think that there are multiple of them.